Spanner/tests/System/QueryTest.php

```diff
@@ -608,4 +668,396 @@ public function arrayTypesNull()
             [Database::TYPE_DATE],
         ];
     }
+
+    /**
+     * @group spanner-query-struct
+     */
+    public function testBindStructParameter()
+    {
+        $db = self::$database;
+
+        $res = $db->execute('SELECT @structParam.userf, @p4', [
+            'parameters' => [
+                'structParam' => [
+                    'userf' => 'bob',
+                    'threadf' => 1
+                ],
+                'p4' => 10
+            ],
+            'types' => [
+                'structParam' => (new StructType)
+                    ->add('userf', Database::TYPE_STRING)
+                    ->add('threadf', Database::TYPE_INT64)
```

Comment:
why can't the type of struct params be inferred?

Reply:
They can be now. I've added tests to prove that case.